### ISSUE TITLE
Pinned view with corner radius for UI sugar.

### ISSUE THREAD

**Issue by takasurazeem** [danielsaidi/ScrollKit#21]
Hi, 

I totally love this library and great work indeed. However, I was trying to add this view in the header and make it sticky so the content go under the rounded corner view as I scroll but it fades away with the header. And I cannot add it to content as it scrolls, can you please guide me on how I can add it effectively so that I can achieve my end goal? I need it for some project and I could really use your help.

Thanks.

![image](https://github.com/user-attachments/assets/5230204d-51e6-4c43-9874-7da772b02151)

**Comment by takasurazeem** [comment]
Trying to achieve something like this.

https://github.com/asvaghasiya/ScrollView-Header-stretching-Animations-Swift/blob/master/ScrollView-Header.gif

**Comment by danielsaidi** [comment]
Hi @takasurazeem 

Happy to hear that you like it :)

Hmmm, since this includes a kind of overlap, I'm not sure the current implementation would work for that.

**Comment by takasurazeem** [comment]
> Hi @takasurazeem 
> 
> 
> 
> Happy to hear that you like it :)
> 
> 
> 
> Hmmm, since this includes a kind of overlap, I'm not sure the current implementation would work for that.

Yeah, I've had same issue trying to get it to work perfectly. But since this design is becoming very common now a days in applications, I was wondering if it could be something built standard into the library.

**Comment by danielsaidi** [comment]
> Yeah, I've had same issue trying to get it to work perfectly. But since this design is becoming very common now a days in applications, I was wondering if it could be something built standard into the library.

That would be really nice :)

**Comment by gabrielribeiro** [comment]
Hey @takasurazeem, maybe adding a negative offset to `content()` like the code below. If you like it, I can open a PR.

```swift
ScrollViewWithOffsetTracking(
    axes,
    showsIndicators: showsIndicators,
    onScroll: handleScrollOffset
) {
    VStack(spacing: 0) {
        scrollHeader
        content()
            .background(.background)
            .clipShape(UnevenRoundedRectangle(topLeadingRadius: 8, topTrailingRadius: 8))
            .frame(maxHeight: .infinity)
            .offset(y: -30)
        }
    }
}
```

![Image](https://github.com/user-attachments/assets/09200a14-544a-4a9a-9bcd-1c040f1fedb3)

**Comment by danielsaidi** [comment]
That looks great @gabrielribeiro 🙌 

What do you think would make most sense - a view modifier like `.roundedScrollHeaderOverflow` that does what you do with the view, or a `ScrollViewWithStickyHeader` configuration/style that has a boolean property for this?

**Comment by gabrielribeiro** [comment]
Thinking about it, I would offer 3 options:
- roundedScrollHeaderOverflow with corner radius parameter. 
- customScrollHeaderOverflow with offset y parameter (so developer can implement whatever he wants with flexibility). 
- a default option as it is today, no overflow, unchanged.
I think it covers everything.

**Comment by danielsaidi** [comment]
@takasurazeem @gabrielribeiro I can confirm that this looks great for the Spotify example screen, although the sticky header behaves a bit strange when it simulates the navbar in this particular case:

<img width="350" alt="Image" src="https://github.com/user-attachments/assets/4984e8f9-a471-4752-9cbc-656c841731bb" />

**Comment by danielsaidi** [comment]
I found an even better way! 🎉 

I have added a new parameter to the scroll view with sticky header, and refactored how the header stretches and shrinks. It should now behave a lot better, and have no risk of overlapping with the underlying header, since that is now always hidden.

Passing in a corner radius will automatically apply it as an inverted mask to the header, after which the content will flow below the header. Passing in <0 or nil will automatically disable the effect.

|||||
|---|---|---|---|
|<img width="713" alt="Image" src="https://github.com/user-attachments/assets/a0d0b33c-fed4-4085-9d8b-276ad5c0a72b" />|
<img width="713" alt="Image" src="https://github.com/user-attachments/assets/27774c0a-4646-4916-9575-e09ef544205c" />|
<img width="713" alt="Image" src="https://github.com/user-attachments/assets/0d2f2e42-b2aa-46e8-a7d8-0748f913e6d6" />|
<img width="713" alt="Image" src="https://github.com/user-attachments/assets/317df820-9da8-4d34-a7b6-f16d310bf241" />|

**Comment by danielsaidi** [comment]
And hey - it even works on macOS! 

<img width="695" alt="Image" src="https://github.com/user-attachments/assets/dfd94ff6-2e3d-4dbc-8e0d-bdd2ad28ea14" />

...and tvOS:

<img width="1753" alt="Image" src="https://github.com/user-attachments/assets/08658a08-d5dc-4095-986f-325e7c957f70" />
